code/drasil-code/lib/Language/Drasil/Code/Imperative/GOOL/ClassInterface.hs

```diff
@@ -52,6 +52,10 @@ type CaseName = String
 type ExamplePurpose = String
 -- | Description of example
 type ExampleDescr = String
+-- | Motivation of example
```

Comment:
While it is good that you are following the established pattern in this code, the problem is that the pattern is wrong... in that GOOL shouldn't know anything at all about 'examples'.

The real problem isn't these types, but `ReadMeInfo` which shouldn't be defined inside `GOOL` at all!

So I think the first step of the implementation, as a separate PR, would be to first fix this and move this information out of `GOOL`.